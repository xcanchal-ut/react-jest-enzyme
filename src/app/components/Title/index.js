import React, { PropTypes } from 'react';

const Title = (props) => (
    <h1>{props.title}</h1>
);

Title.propTypes = {
    title: PropTypes.string.isRequired,
};

export default Title;
