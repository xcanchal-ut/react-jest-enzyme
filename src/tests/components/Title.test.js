import React from 'react';
import { shallow } from 'enzyme';
import Title from '../../app/components/Title/';

// Shallow rendering w/snapshot
it('Should render an h1 with the "Apply filter" text', () => {
    const wrapper = shallow(
        <Title title="Apply filter" />
    );
    expect(wrapper).toMatchSnapshot();
});
