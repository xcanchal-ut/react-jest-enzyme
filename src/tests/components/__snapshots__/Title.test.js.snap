// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render an h1 with the "Apply filter" text 1`] = `
<h1>
  Apply filter
</h1>
`;
